syft/messaging/plan/procedure.py

```diff
@@ -74,16 +74,30 @@ def update_ids(
             from_worker: The previous worker that built the plan.
             to_worker: The new worker that is running the plan.
         """
+
+        # We operate on simplified content of Operation, hence all values should be simplified
+        from_workers_simplified = None
+        to_workers_simplified = None
+        if from_worker and to_worker:
```

Comment:
Q: We will always have ```from_worker``` and ```to_worker```. Is there any case only one is ```!= None```?

Reply:
I think replacement doesn't make sense when any of them are `None`.
This was existing code, I just moved it out of the loop to avoid making simplification each time.